### PR TITLE
NVSHAS-9700/NVSHAS-9699 Retool logic for parseDotNetPackage

### PR DESCRIPTION
**Pull request by Acmarr**
Fixes an issue where module count and contents could be inconsistent. Added a new unit test to cover this as well as updating  TestParseDotNetModuleNameVersion test.

### CODE REVIEW

File: share/scan/apps.go [hunk 1]
```diff
@@ -690,15 +690,34 @@ func (s *ScanApps) parseDotNetPackage(filename string, fullpath string) {
...
 
-	var coreVersion string
+	pkgs := parseDotNetJsonData(filename, fullpath, dotnet)
+
+	if len(pkgs) > 0 {
+		s.pkgs[filename] = pkgs
+	}
+
+	fmt.Printf("%v Entries: %v\n", filename, len(s.pkgs[filename]))
+	// for _, pkg := range s.pkgs[filename] {
```
Comment (jayhuang-suse) [inline share/scan/apps.go:715]: Please remove this debug logs

File: share/scan/apps.go [hunk 2]
```diff
@@ -690,15 +690,34 @@ func (s *ScanApps) parseDotNetPackage(filename string, fullpath string) {
...
 
-	if data, err := os.ReadFile(fullpath); err != nil {
+	data, err := os.ReadFile(fullpath)
+
+	if err != nil {
 		log.WithFields(log.Fields{"err": err, "fullpath": fullpath, "filename": filename}).Error("Failed to read file")
 		return
-	} else if err = json.Unmarshal(data, &dotnet); err != nil {
+	}
+	// if filename == "app/Business.ProfileService.deps.json" {
```
Comment (jayhuang-suse) [inline share/scan/apps.go:699]: Please remove this debug logs

File: share/scan/apps.go [hunk 3]
```diff
@@ -719,34 +738,11 @@ func (s *ScanApps) parseDotNetPackage(filename string, fullpath string) {
...
-		if token == "Microsoft.NETCore.App" || token == "Microsoft.AspNetCore.App" {
-			if i < len(tokens)-1 {
-				coreVersion = tokens[i+1]
-			}
-			break
-		}
-	}
-
 	if targets, ok := dotnet.Targets[dotnet.Runtime.Name]; ok {
+		//fmt.Printf("targets len: %v\n", len(targets))
```
Comment (jayhuang-suse) [inline share/scan/apps.go:742]: Please remove this debug logs

**Review by jayhuang-suse — CHANGES_REQUESTED** [review]
Please use the commit title within 50 characters.

File: share/scan/apps.go (hunk 3 above)
Comment (Acmarr) [reply share/scan/apps.go]: Removed the remaining debug lines.

**Review by jayhuang-suse — APPROVED** [review]
Look good to me

**Pull request merged**